src/main/java/org/mifos/processor/bulk/camel/routes/ProcessorStartRoute.java

```diff
@@ -143,7 +148,73 @@ private void setup() {
             exchange.setProperty(TENANT_NAME, tenantName);
         }).setHeader("Content-Type", constant("application/json;charset=UTF-8")).log("Completed route direct:validate-tenant");
 
-        from("direct:start-batch-process-csv").id("direct:start-batch-process-csv").log("Starting route direct:start-batch-process-csv")
+        // this route is responsible for editing the incoming records based on configuration
+        // this step is done to make sure the file format of CSV is not altered and only the data is updated based on config
+        from("direct:update-incoming-data")
+                .id("direct:update-incoming-data")
+                .log("direct:update-incoming-data")
+                // [LOCAL_FILE_PATH] is already set in [direct:validateFileSyncResponse] route
+                .to("direct:get-transaction-array")
+                // make sure new data is set under the exchange variable [RESULT_TRANSACTION_LIST]
+                .process(exchange -> {
+                    String registeringInstituteId = exchange.getProperty(REGISTERING_INSTITUTE_ID, String.class);
+                    String programId = exchange.getProperty(PROGRAM_ID, String.class);
+                    logger.debug("Inst id: {}, prog id: {}", registeringInstituteId, programId);
+                    if (registeringInstituteId == null || registeringInstituteId.isEmpty() ||
```

Comment:
@danishjamal104 You can use `StringUtils.isEmpty()`. It does both, null and empty string check is more readable manner.